src/lib/Microsoft.Health.Extensions.Fhir.R4/Telemetry/Exceptions/FhirServiceExceptionProcessor.cs

```diff
@@ -51,11 +50,13 @@ public static (Exception customException, string errorName) CustomizeException(E
                             errorName = nameof(FhirServiceErrorCode.AuthorizationError);
                             return (new UnauthorizedAccessFhirServiceException(message, exception, helpLink, errorName), errorName);
                         case HttpStatusCode.NotFound:
-                            message = FhirResources.FhirServiceNotFound;
+                            message = DetermineNotFoundErrorMessage(exception);
                             errorName = nameof(FhirServiceErrorCode.ConfigurationError);
                             return (new InvalidFhirServiceException(message, exception, errorName), errorName);
-                        default:
+                        case HttpStatusCode.BadRequest:
```

Comment:
What scenario gives a `BadRequest`? I'm wondering if it needs a custom exception like `InvalidFhirServiceException` to indicate it's a user error, but based on these changes I'm thinking it's more likely that this case can be removed and handled as part of the `default` case. (Please see my next comment.)

Reply:
Thanks for pointing this out. I pushed new changes to the FhirServiceExceptionProcessor. Please see my comment below